dazel/lib/src/config/config_set.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:async';
+
+import '../bazelify/pubspec.dart';
+import 'build_config.dart';
+
+/// The [BuildConfig]s for a package and it's dependencies.
+class BuildConfigSet {
+  final BuildConfig local;
+  final Map<String, BuildConfig> dependencies;
+
+  static Future<BuildConfigSet> forPackages(
+          String localPackagePath,
+          Map<String, String> dependencyPaths,
+          Map<String, Pubspec> dependencyPubspecs) async =>
+      new BuildConfigSet(await _readLocalBuildConfig(localPackagePath),
+          await _readBuildConfigs(dependencyPaths, dependencyPubspecs));
+
+  BuildConfigSet(this.local, this.dependencies);
+}
+
+Future<Map<String, BuildConfig>> _readBuildConfigs(
```

Comment:
This probably deserves a comment describing the return type (specifically that the keys are package paths).